tests/test_cleaneval.py

```diff
@@ -0,0 +1,486 @@
+    with open(filename, 'rb') as src_file:
+        detection = chardet.detect(src_file.read())
+        src_file.close()
+    if detection['encoding'] != 'ascii':
+        encoding = 'ISO 8859-9'
+    elif detection['encoding'] == 'Windows-1254':
+        encoding = 'ISO 8859-1'
+    else:
+        encoding = detection['encoding']
+    return encoding
+
+def __check_encoding(path):
+    files = os.listdir(path)
+    for filename in files:
+        file_path = path+'/'+filename
+        encoding = __get_encoding(file_path)
+        if encoding != 'utf-8':
+            print(encoding)
+
+re_URL = re.compile(r'^\s*URL.*$', re.MULTILINE)
+re_TAG = re.compile(r'(<[phl]>)', re.IGNORECASE)
+re_WS = re.compile(r'\s+')
+re_CTRL = re.compile(r'[\x00-\x1F]+')
+re_HI = re.compile(r'[\x80-\xFF]+')
+re_CLEAN_URL = re.compile(r'^\s*##.*$', re.MULTILINE) #For frsv gold dataset
+
+def normalize(text, ascii=False, unlabelled=False):
+    text = re_URL.sub("", text)           # remove URL line at start of gold standard files
+    text = re_CLEAN_URL.sub("", text)
+    text = re_CTRL.sub(" ", text)         # replace any control characters by spaces (includes newlines)
+    if unlabelled:
+        text = re_TAG.sub(r'\n<p> ', text) # start each segment on new line, normalise tags
+    else:
+        text = re_TAG.sub(r'\n\g<1> ', text)  # only break lines before segment markers
+        text = re_WS.sub(" ", text)           # normalise whitespace (including line breaks) to single spaces
+        if ascii:
+            text = re_HI.sub("", text)        # delete non-ASCII characters (to avoid charset problems)
+        return text
+
+## return diff as list of tuples ("equal"/"insert"/"delete", [text words], [gold words])
+def make_diff(alignment, text_w, gold_w):
+	diff = []
+	for tag, i1, i2, j1, j2 in alignment.get_opcodes():
+		text_region = text_w[i1:i2]
+		gold_region = gold_w[j1:j2]
+		if tag == "replace":
+			diff.append( ("delete", text_region, []) )
+			diff.append( ("insert", [], gold_region) )
+		else:
+			diff.append( (tag, text_region, gold_region) )
+	return diff
+
+## return evaluation measures for given diff:
+##   (f-score, precision, recall, [labelled] segmentation f-score, precision, recall)
+def evaluate(diff):
+    tp = fp = fn = 0
+    tag_tp = tag_fp = tag_fn = 0
+    for tag, text, gold in diff:
+        text_tags = len(list(filter(re_TAG.match, text)))
+        gold_tags = len(list(filter(re_TAG.match, gold)))
+        text_l = len(text)
+        gold_l = len(gold)
+        if tag == "delete":
+            fp += text_l
+            tag_fp += text_tags
+        elif tag == "insert":
+            fn += gold_l
+            tag_fn += gold_tags
+        else:
+            tp += text_l
+            tag_tp += text_tags
+            assert text_l == gold_l
+            assert text_tags == gold_tags
+    result = __evaluate(tp, fp, fn, tag_tp, tag_fp, tag_fn)
+    return result
+
+def __evaluate(tp, fp, fn, tag_tp, tag_fp, tag_fn):
```

Comment:
it's not clear what the difference between `evaluate` and `__evaluate` is; my guess is that the former is a function you wrote and the latter is from the original cleantext?  

It also looks like these functions are only used in one place: at the end of the `do_cleaneval` and `do_jaccard` functions; therefore, these functions should be moved so they are located after those functions; this will make it easier to read the code, since you don't have to scroll to find the functions while reading `do_*`... the next block of code you need to read is just right there.

It's also not clear to me why they should be separate functions, so I might just put them inside the `do_*` functions to begin with and remove these functions.

If you don't put them inside the other functions, you need to rename them more appropriately.